src/coreclr/vm/precode.h

```diff
@@ -607,11 +607,27 @@ class Precode {
 };
 
 // Verify that the type for each precode is different
-static_assert_no_msg(StubPrecode::Type != NDirectImportPrecode::Type);
-static_assert_no_msg(StubPrecode::Type != FixupPrecode::Type);
-static_assert_no_msg(StubPrecode::Type != ThisPtrRetBufPrecode::Type);
+
+#if defined(HAS_NDIRECT_IMPORT_PRECODE) && defined(HAS_FIXUP_PRECODE)
```

Comment:
A nit - all of these defines that you've added checks for here are always defined for all architectures. I'd prefer not adding these checks here and remove all of those defines instead in a separate cleanup (I would do that myself)